table/block_based_table_reader.cc

```diff
@@ -196,8 +197,9 @@ class PartitionIndexReader : public IndexReader, public Cleanable {
     }
     return NewTwoLevelIterator(
         new BlockBasedTable::BlockEntryIteratorState(
-            table_, ReadOptions(), skip_filters, is_index, block_cache_cleaner),
-        index_block_->NewIterator(comparator_, nullptr, true));
+            table_, ReadOptions(), icomparator_, skip_filters, is_index,
```

Comment:
`table_` contains the comparator (`table_->rep_->internal_comparator`). `BlockBasedTable::BlockEntryIteratorState()` could get it from there instead of taking it as another argument.